pkg/gce-csi-driver/node.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gceGCEDriver
+
+import (
+	"fmt"
+	"os"
+
+	csi "github.com/container-storage-interface/spec/lib/go/csi/v0"
+	"github.com/golang/glog"
+	mountmanager "github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pkg/mount-manager"
+	utils "github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pkg/utils"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type GCENodeServer struct {
+	Driver  *GCEDriver
+	Mounter mountmanager.Mounter
+}
+
+func (ns *GCENodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
```

Comment:
Publish and Unpublish should add some locking in case it gets called multiple times for the same mount.  Kubernetes theoretically synchronizes this for the plugin, but it's not a requirement in the CSI spec.

Reply:
And similarly for Stage/Unstage

Reply:
just locked all of the node calls for now, I'll add a TODO to make them more fine grained and only lock mutually exclusive calls with each other in the future